### PR TITLE
Introduce environments index page to list all the environments.

### PR DESCRIPTION
**Pull request by GaneshSPatil**
* List all the environments in expand/collapsible manner on the environments list page
* Show information about number of pipelines and number of agents
* Retain environment specific show page for the redirection from the dashboard
* Move environment next to pipelines under admin's dropdown
* Add redirects from /go/environments to /go/admin/environments

This PR addresses all the features from: https://github.com/gocd/gocd/pull/5113 PR

### CODE REVIEW

**Comment by GaneshSPatil** [comment]
screen shot 2018-10-05 at 5 35 48 pm
screen shot 2018-10-05 at 5 35 51 pm
screen shot 2018-10-05 at 5 36 14 pm

File: server/webapp/WEB-INF/rails/app/assets/stylesheets/new-theme.scss [hunk 1]
```diff
@@ -83,6 +83,58 @@ h4 {
...
+  cursor:        pointer;
+  padding:       10px 50px 10px 30px;
+  margin:        0 15px;
+  background:    #fff;
+  border:        1px solid #ddd !important;
+  border-radius: 5px;
+
+  &:after {
+    content:     "\f105";
+    font-family: 'FontAwesome';
```
Comment (akshaydewan) [inline server/webapp/WEB-INF/rails/app/assets/stylesheets/new-theme.scss]: Why not use the `icon-after` mixin?

**Comment by GaneshSPatil** [comment]
Old code.. can we use mixin here too? If so i can change this code

**Comment by akshaydewan** [comment]
> Old code.. can we use mixin here too? If so i can change this code
Ok, I got thrown off my the name `new-theme.scss`

File: server/webapp/WEB-INF/rails/config/routes.rb [hunk 2]
```diff
@@ -187,19 +187,22 @@
     get "templates/:name" => "config_view/templates#show", as: :config_view_templates_show, constraints: {name: TEMPLATE_NAME_FORMAT}
   end
 
-  scope 'environments' do
+  get 'environments', to: redirect('admin/environments'), as: :environment_redirect
```
Comment (bdpiparva) [inline server/webapp/WEB-INF/rails/config/routes.rb]: Currently `/go/environments` is redirected to => `/go/admin/environments`.

Need to consider redirects for the following routes -

- `go/environments/foo/show` =>`go/admin/environments/foo/show`
- `go/environments/new` => `go/admin/environments/new`
- POST  to `/go/environments/:env_name` => `/go/admin/environments/:env_name`

Comment (bdpiparva) [reply server/webapp/WEB-INF/rails/config/routes.rb]: This is to make sure that old links for env page will continue to work. We will put deprecation of old routes in the release note.

**Comment by bdpiparva** [comment]
Changed HTML structure and css to make the UI consistent with other pages

screen shot 2018-10-09 at 6 12 46 pm

**Pull request merged**